### PR TITLE
UefiPayloadPkg/Library/PlatformBootManagerLib: Handle Primary Video

### PR DESCRIPTION
**Pull request by walterchris**
Based on what has been read from the BoardSettings we set the primary
video output (one and only GOP). All other GOPs are uninstalled as it is
unclear how Windows selects the "first" GOP.

Signed-off-by: Christian Walter <christian.walter@9elements.com>

### CODE REVIEW

File: UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c [hunk 1]
```diff
@@ -211,6 +224,159 @@ ConnectRootBridge (
...
+
+  // Fetch Board Settings
+  Status = gRT->GetVariable(BOARD_SETTINGS_NAME,
+    &gEfiBoardSettingsVariableGuid,
+    NULL,
+    &BoardSettingsSize,
+    &BoardSettings);
+
+  if (EFI_ERROR(Status)) {
+    DEBUG((EFI_D_ERROR, "Fetching Board Settings errored with %x\n", Status));
```
Comment (sylv-io) [inline UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c:284]: PatchCheck: EFI_D_ERROR was used, but DEBUG_ERROR is now recommended

Comment (sylv-io) [inline UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c:284]: PatchCheck: EFI_D_ERROR was used, but DEBUG_ERROR is now recommended

File: UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c [hunk 2]
```diff
@@ -211,6 +224,159 @@ ConnectRootBridge (
...
+
+  // Locate all GOPs
+  Status = gBS->LocateHandleBuffer(ByProtocol, 
+                  &gEfiGraphicsOutputProtocolGuid,
+                  NULL,
+                  &HandleCount,
+                  &Handle);
+
+  if (EFI_ERROR(Status)) {
+    DEBUG((EFI_D_ERROR, "Fetching handles errored with %x\n", Status));
```
Comment (sylv-io) [inline UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c:296]: PatchCheck: EFI_D_ERROR was used, but DEBUG_ERROR is now recommended

File: UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c [hunk 3]
```diff
@@ -211,6 +224,159 @@ ConnectRootBridge (
...
+
+    // Check which GOP should be enabled
+    if (StringCompareLen(Str, GRAPHICS_KVM_OUTPUT, StrLen(GRAPHICS_KVM_OUTPUT)))  {
+      DEBUG ((EFI_D_INFO, "Found the KVM Device.."));
+      // If Primary Video not KVM - disable.
+      if (BoardSettings.PrimaryVideo != 0) {
+        DEBUG ((EFI_D_INFO, "Disabling"));
+        Status = gBS->UninstallProtocolInterface(Handle[Index], &gEfiGraphicsOutputProtocolGuid, mGop);
+        if (EFI_ERROR (Status)) {
+          DEBUG((EFI_D_ERROR, "Uninstalling Handle errored with %x\n", Status));
```
Comment (sylv-io) [inline UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c:323]: PatchCheck: EFI_D_ERROR was used, but DEBUG_ERROR is now recommended.

Maybe you could change all appearing of `EFI_D_ERROR` in this file.

File: UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c (hunk 1 above)
Comment (sylv-io) [reply UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c:284]: I removed PatchCheck from the CI

**Comment by sylv-io** [comment]
? :smile:

**Comment by walterchris** [comment]
Let's not talk about it. :)

**Pull request closed without merging**